app_test.go

```diff
@@ -1,5 +1,72 @@
 package buffalo
 
+import (
+	"net/http"
+	"testing"
+	"time"
+)
+
 func voidHandler(c Context) error {
 	return nil
 }
+
+func TestGracefulShutdown(t *testing.T) {
+	requestFinished := false
+
+	app := New(NewOptions())
+	app.GET("/slow", func(c Context) error {
+		time.Sleep(4 * time.Second)
+		requestFinished = true
+		return nil
+	})
+
+	go func() {
+		time.Sleep(2 * time.Second)
+		app.Stop(nil)
+	}()
+
+	go func() {
+		time.Sleep(1 * time.Second)
+		_, err := http.Get("http://localhost:17171/slow")
+		if err != nil {
+			t.Error("unexpected error requesting slow endpoint", err)
+		}
+	}()
+
+	app.Start(":17171")
```

Comment:
Not a fan of the hard coded port here. Perhaps we can do what they do with `httptest.NewServer` https://golang.org/pkg/net/http/httptest/#NewServer

Reply:
Me either.  From a quick dive into NewServer and google looks like using `:0` to get a dynamic port is the way to go but will require creating the Listener to pull it later.

https://golang.org/src/net/http/httptest/server.go?s=2300:2344#L55
https://stackoverflow.com/questions/31632506/how-to-get-address-for-http-listenandserve
https://stackoverflow.com/questions/43424787/how-to-use-next-available-port-in-http-listenandserve